source/extensions/wasm/config.cc

```diff
@@ -21,7 +21,7 @@ Server::WasmSharedPtr WasmFactory::createWasm(const envoy::config::wasm::v2::Was
   // Create a base WASM to verify that the code loads before setting/cloning the for the individual threads.
   auto base_wasm =
       Common::Wasm::createWasm(config.id(), config.vm_config(), context.clusterManager(),
-                               context.dispatcher(), context.api());
+                               context.dispatcher(), context.api(), *context.scope(), context.scope());
```

Comment:
This createWasm is confusing. It should be either pass the ownership or not.

Reply:
The ownership is shared because that is the API for Scope.  I could create a separate createWasm which just takes the Scope pointer, but I thought that it wasn't worth the maintainance overhead.

Reply:
What I am missing here is the concept of `Scope`. The merge shouldn't be blocked by my comment.